### PR TITLE
Fix more gometalinter errors and integrate with CircleCI

### PR DESCRIPTION
**Pull request by fern4lvarez**
@otoolep, please double check if `circle.yml` file looks fine.

It's required to run gometalinter on all subpackages separately as it doesn't support recursion (see https://github.com/alecthomas/gometalinter/issues/12).

Please see commit messages to find which gometalinter issues are the commits themselves fixing.

### CODE REVIEW

File: system_test.go [hunk 1]
```diff
@@ -72,10 +72,10 @@ func createCluster(t *testing.T, numNodes int, host string, basePort int, path s
 	// Create remaining nodes in cluster.
 	for i := 1; i < numNodes; i++ {
 		port := basePort + i
-		nodePath := filepath.Join(path, strconv.Itoa(i))
```
Comment (otoolep) [inline system_test.go]: This doesn't look right.

File: system_test.go [hunk 2]
```diff
@@ -72,10 +72,10 @@ func createCluster(t *testing.T, numNodes int, host string, basePort int, path s
...
+		nodePath = filepath.Join(path, strconv.Itoa(i))
 		mustMkDirAll(nodePath)
 
-		s := server.NewServer(nodePath, "db.sqlite", 100000, host, port)
+		s = server.NewServer(nodePath, "db.sqlite", 100000, host, port)
```
Comment (otoolep) [inline system_test.go]: Nor this.

File: server/server.go [hunk 3]
```diff
@@ -481,13 +481,7 @@ func (s *Server) writeHandler(w http.ResponseWriter, req *http.Request) {
 
 	currentIndex := s.raftServer.CommitIndex()
 	count := currentIndex - s.snapConf.lastIndex
-	if uint64(count) > s.snapConf.snapshotAfter {
```
Comment (otoolep) [inline server/server.go]: Is moving this to a separate function a lint thing?

**Comment by otoolep** [comment]
The build is red, so I'll take another look when it passes CI.

File: system_test.go (hunk 1 above)
Comment (fern4lvarez) [inline system_test.go]: See commit message:
`system_test.go:75:3:warning: Shadowing variable `nodePath` (go-nyet)`

File: system_test.go (hunk 2 above)
Comment (fern4lvarez) [inline system_test.go]: See commit message:
`system_test.go:78:3:warning: Shadowing variable `s` (go-nyet)`

File: server/server.go (hunk 3 above)
Comment (fern4lvarez) [inline server/server.go]: See commit message:
`server/server.go:469:1:warning: cyclomatic complexity 11 of
function (*Server).writeHandler() is high (> 10) (gocyclo)`

It's about the complexity of the function. Extracting this method reduces it under 10, but any kind of refactoring would have done the job as well.

File: system_test.go (hunk 1 above)
Comment (otoolep) [inline system_test.go]: D'oh. You're right.

File: server/server.go [hunk 4]
```diff
@@ -620,3 +614,13 @@ func (s *Server) leaderRedirect(w http.ResponseWriter, r *http.Request) {
 	}
 	http.Redirect(w, r, u+r.URL.Path, http.StatusTemporaryRedirect)
 }
+
+func (s *Server) handleSnapshotThreshold(count, index uint64) {
+	if count > s.snapConf.snapshotAfter {
```
Comment (otoolep) [inline server/server.go]: I'd prefer if this check was moved outside of the function -- only call it if count is greater than the threshold. It's easier to follow the code that way, as well as explicitly uint-test the function.

Comment (fern4lvarez) [inline server/server.go]: sure thing!

**Comment by fern4lvarez** [comment]
Closing for now, having some issues on running gometalinter on all subpackages.

**Pull request closed without merging**